conans/client/tools/net.py

```diff
@@ -9,10 +9,10 @@
 _global_requester = None
 
 
-def get(url, md5='', sha1='', sha256='', destination="."):
+def get(url, md5='', sha1='', sha256='', destination=".", filename=""):
     """ high level downloader + unzipper + (optional hash checker) + delete temporary zip
     """
-    filename = os.path.basename(url)
+    filename = filename or os.path.basename(url)
```

Comment:
Check if filename starts with "?" and raise an exception indicating that the "filename" parameter should be used.

Reply:
And add a test.